chain/chain/src/chain.rs

```diff
@@ -2891,97 +2942,159 @@ impl<'a> ChainUpdate<'a> {
+                            block_timestamp,
+                            &prev_block_hash,
+                            &block_hash,
                             &receipts,
                             chunk.transactions(),
                             chunk_inner.validator_proposals(),
-                            prev_block.header().gas_price(),
+                            gas_price,
                             gas_limit,
-                            &block.header().challenges_result(),
-                            *block.header().random_value(),
+                            &challenges_result,
+                            random_seed,
                             true,
                             is_first_block_with_chunk_of_version,
                             #[cfg(feature = "sandbox")]
-                            self.states_to_patch.take(),
+                            states_to_patch,
                             #[cfg(not(feature = "sandbox"))]
                             None,
-                        )
-                        .map_err(|e| ErrorKind::Other(e.to_string()))?;
-
-                    let (outcome_root, outcome_paths) =
-                        ApplyTransactionResult::compute_outcomes_proof(&apply_result.outcomes);
-
-                    self.chain_store_update.save_trie_changes(apply_result.trie_changes);
-                    // Save state root after applying transactions.
-                    self.chain_store_update.save_chunk_extra(
-                        &block.hash(),
-                        shard_id,
-                        ChunkExtra::new(
-                            &apply_result.new_root,
-                            outcome_root,
-                            apply_result.validator_proposals,
-                            apply_result.total_gas_burnt,
-                            gas_limit,
-                            apply_result.total_balance_burnt,
-                        ),
-                    );
-                    self.chain_store_update.save_outgoing_receipt(
-                        &block.hash(),
-                        shard_id,
-                        apply_result.receipt_result,
-                    );
-                    // Save receipt and transaction results.
-                    self.chain_store_update.save_outcomes_with_proofs(
-                        &block.hash(),
-                        shard_id,
-                        apply_result.outcomes,
-                        outcome_paths,
-                    );
+                        ) {
+                            Ok(apply_result) => {
+                                Ok(ApplyChunkResult::SameHeight(SameHeightResult {
+                                    shard_id,
+                                    gas_limit,
+                                    apply_result,
+                                }))
+                            }
+                            Err(err) => Err(ErrorKind::Other(err.to_string()).into()),
```

Comment:
Curisous, why this isn't just `Err(err) => Err(err),`?

Reply:
I am as well, but this is to preserve behavior